### ISSUE TITLE
What if the reward is 0 when starting the training?

### ISSUE THREAD

**Issue by eduOS** [ruotianluo/self-critical.pytorch#4]
Hi, Dr. Luo,

Hats off to you for this awesome repository. I have translated it into TensorFlow. 

But I find that the rewards are all 0's. Do you have any strategies to prevent the rewards being 0's? At the beginning of the training it seems reasonable that the rewards will be small, and even to the extreme of being 0(no overlap both between the target sequence and sampled sequence, and between the target sequence and the greedy sequence), then the delta will be also 0's for the weights. 

Does my concern make sense?
Best, 
Lerner

**Comment by eduOS** [comment]
I got to know that the XE should be added whether in the pretraining or in the training process.

**Comment by ruotianluo** [comment]
The rewards shouldn't be all zero.

**Comment by eduOS** [comment]
@ruotianluo Why are you so certain? Am I missing something? Methinks at the beginning both the sampled sequence and the greedy sequence would likely be totally different from the ground truth one, and hence the reward is 0 minus 0.

**Comment by ruotianluo** [comment]
You need to pertrain using cross entropy loss first.

**Comment by eduOS** [comment]
Yeah, you are right. Thanks. I add the cross entropy(with a ratio of, say 0.1) in the loss.
